internal/commands/accesslist/update_test.go

```diff
@@ -0,0 +1,207 @@
+package accesslist
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+)
+
+func TestAllowedIPUpdateHandler(t *testing.T) {
+	projectID := "projectID"
+	appID := "appID"
+	app := realm.App{
+		ID:          appID,
+		GroupID:     projectID,
+		ClientAppID: "test-abcde",
+		Name:        "test",
+	}
+
+	allowedIPs := []realm.AllowedIP{
+		{ID: "address1", Address: "0.0.0.0", Comment: "comment"},
+		{ID: "address2", Address: "192.1.1.1"},
+		{ID: "address3", Address: "192.158.1.38", Comment: "cool comment"},
+	}
+
+	for _, tc := range []struct {
+		description    string
+		testAddress    string
+		testNewAddress string
+		testComment    string
+	}{
+		{
+			description:    "should return a successful message for a successful update for address and comment",
+			testAddress:    "0.0.0.0",
+			testNewAddress: "2.2.2.2",
+			testComment:    "new comment",
+		},
+		{
+			description:    "should return a successful message for an update with only an address",
+			testAddress:    "192.1.1.1",
+			testNewAddress: "68.192.33.2",
+		},
+		{
+			description: "should return a successful message for an empty update",
+			testAddress: "192.158.1.38",
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			realmClient := mock.RealmClient{}
+
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				return []realm.App{app}, nil
+			}
+			realmClient.AllowedIPsFn = func(groupID, appID string) ([]realm.AllowedIP, error) {
+				return allowedIPs, nil
+			}
+			realmClient.AllowedIPUpdateFn = func(groupID, appID, allowedIPID, newAddress, comment string) error {
+				return nil
+			}
+
+			cmd := &CommandUpdate{updateInputs{
+				cli.ProjectInputs{projectID, appID, nil},
+				tc.testAddress,
+				tc.testNewAddress,
+				tc.testComment,
+			}}
+
+			out, ui := mock.NewUI()
+
+			assert.Nil(t, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+
+			assert.Equal(t, "Successfully updated allowed IP\n", out.String())
+		})
+	}
+
+	t.Run("should return an error", func(t *testing.T) {
+		for _, tc := range []struct {
+			description string
+			inputs      updateInputs
+			clientSetup func() realm.Client
+			expectedErr error
+		}{
+			{
+				description: "if there is no app",
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return nil, errors.New("Something went wrong with the app")
```

Comment:
[ubernit] i know its just test code, but errors shouldn't be starting with capital letters, might as well adjust that the few times it occurs here